content/updates/understanding-the-terraform-check-block-feature.md

```diff
@@ -0,0 +1,210 @@
+* As `terraform plan` and `terraform apply` represent different stages in the workflow, _the purpose of checks can also diverge into post-plan, post-apply, and the ones relevant for both cases_. We see great potential for improvement here, such as the possibility of labeling or ignoring checks for a particular stage so that checks could be built-in as smoothly as possible.
+
+In addition to that we've faced some limitations with data source usage in the assertsions.
+
+* _Approval is required for `terraform apply`_
+
+  In case of a successful assertion, Terraform requires approval for every `apply` operation due to a configuration reload needed to verify a check block, e.g.:
+
+  ```sh
+  Terraform will perform the following actions:
+
+    # module.tailscale_subnet_router.data.tailscale_device.default will be read during apply
+    # (config will be reloaded to verify a check block)
+  <= data "tailscale_device" "default" {
+        + addresses = [
+            + "100.100.48.62",
+          ]
+        + id        = "8181818181818181"
+        + name      = "mp-automation-tailscale-subnet-router.cat-crocodile.ts.net"
+        + tags      = [
+            + "tag:mp-automation-tailscale-subnet-router",
+          ]
+        + wait_for  = "30s"
+      }
+
+  Plan: 0 to add, 0 to change, 0 to destroy.
+
+  Do you want to perform these actions?
+    Terraform will perform the actions described above.
+    Only 'yes' will be accepted to approve.
+
+    Enter a value:
+  ```
+
+* _Multiple data resource blocks are not supported_
+
+  Unfortunately, it's possible to define only one data source per check at the moment. Otherwise, the error will be thrown:
+
+  ```sh
+  This check block already has a data resource defined at main.tf:83,3-27.
+  ```
+
+  This limitation restrains creating complex assertions within one `check` block.
+
+## Final Thoughts
+
+While intriguing on paper, Terraform's new check block feature has yet to live up to our initial expectations fully. Its practical integration into our Terraform code has revealed some limitations, and it felt less revolutionary than we'd hoped. There were a few hitches like it being equally tied to the plan and apply lifecycle and not being able to flag a check block as a critical fail-safe.
+
+We realize its effectiveness may vary across different scenarios and contexts. It's easy to implement and integrate into your existing infrastructure. While it's unlikely to replace testing tools and strategies completely, it can undoubtedly bear the burden in some cases, especially considering potential improvements in the future. We'll keep this feature in our toolkit as we refine our infrastructure code.
+
+That being said, there's still a good deal of progress to be made in Terraform's testing and validation arena. We recommend exploring and leveraging this feature and look forward to hearing feedback and thoughts from the community!
+
```

Comment:
```suggestion

### Update: New `test` command command in Terraform v1.6

As we were wrapping up this blog post, the Terraform team has released their preview of v1.6 Terraform and announced that it will primarily add a new `terraform test` command that we're very excited for! [More information can be found here](https://discuss.hashicorp.com/t/request-for-testing-terraform-test/56254) and we look forward to testing this out and writing about it in an upcoming Masterpoint post! 

```

Reply:
Great idea! I've edited it a bit to avoid using `Terraform` twice, if you don't mind.

Reply:
Nice!